### PR TITLE
Apis masha

### PR DESCRIPTION
**Pull request by maxaalexeeva**
This is both @BeckySharp 's apis branch and my svo grounder. 
The svo grounder is now available both through a **separate endpoint** (_groundMentionsToSVO_, documented in the Text Reading section of the wiki), producing a json string like this:

```
 {
            "variable": "LAI",
            "groundings": [
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Body",
                    "score": [
                        2
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Phenomenon",
                    "score": [
                        2
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Spatial Phenomenon",
                    "score": [
                        2
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "crop~mature~dry",
                    "className": "Phenomenon",
                    "score": [
                        13
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "(land_allocation)@medium_crop",
                    "className": "Medium Phenomenon",
                    "score": [
                        27
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "(land_allocation)@medium_crop",
                    "className": "Phenomenon",
                    "score": [
                        27
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "(crop_planting)@reference~from_crop",
                    "className": "Phenomenon",
                    "score": [
                        33
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "(crop_planting)@reference~from_crop",
                    "className": "Reference Phenomenon",
                    "score": [
                        33
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "vegetation_canopy",
                    "className": "Body",
                    "score": [
                        13
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "vegetation_canopy",
                    "className": "Phenomenon",
                    "score": [
                        13
                    ],
                    "source": "SVO"
                }
            ]
        }
```
 or as **part of the Aligner** (I check the results using the ExtractAndAlign app), producing a separate link hypothesis like this: 

```
{
		"element_1": {
			"type": "text_span",
			"source": "text_file",
			"content": "S",
			"content_type": "null"
		},
		"element_2": {
			"type": "svo_grounding",
			"source": "svo_ontology",
			"content": {
				"osv_term": "population",
				"class_name": "Process",
				"source": "SVO"
			}
		},
		"score": 2
	}
```

The score for svo grounding is just edit distance, so lower is better. @pauldhein, let me know if that does not work well for your purposes. 
There are probably still minor things to change to make this better, and, @BeckySharp, I am adding you as a reviewer for when you are back, but @pauldhein, try it out and I think it should be safe to merge if that makes things easier on your end.

### CODE REVIEW

**Comment by maxaalexeeva** [comment]
Also, @pauldhein, you mentioned you are using the _pdf_to_mentions_ endpoint, is that right? If yes, Paul and @BeckySharp, do you want the svo grounder in this endpoint as well or should Paul just use the groundMentionsToSVO endpoint after getting mentions from _pdf_to_mentions_?

**Comment by BeckySharp** [comment]
also re the score being lower is better, if it simplifies things on Paul’s
end we can do 1/edit distance too?

**Comment by pauldhein** [comment]
@maxaalexeeva thank you for getting this PR put together so quickly! As far as the end points go, I can certainly use the `groundMentionsToSVO` after using the `pdf_to_mentions` endpoint, and I would like to keep them separate for now. My rationale for this is that I may want to aggregate a set of mentions from multiple subsequent calls to `pdf_to_mentions`, filter them, and then send the resultant set to `groundMentionsToSVO `.

@BeckySharp thank you for drawing my attention to the score we are using. I agree that using just the edit distance for the score is less than ideal. I would like the score to be normalized with regard to the lengths of the strings being compared, and ideally I'd like it to follow the scale of `[0, 1]` where one represents a perfect match, and zero represents nothing in common.

I'm assuming the edit distance we are using is the Levenshtein distance or a close counterpart, is that correct? If that is the case, can we use the following as our score for two strings A and B:
```python
W = max(|A|, |B|)
S = (W - Levenshtein_dist(A, B)) / W
```

I recognize that this isn't a full metric because it violates the triangle property, but for now I only need to make ordinal comparison amongst the scores, so I think we can wait on making the score into a full metric. In the future we can use [this](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=4160958) as our full metric score.

File: text_reading/src/main/scala/org/clulab/aske/automates/alignment/Aligner.scala [hunk 1]
```diff
@@ -15,7 +22,7 @@ trait Aligner {
 }
 
 
-class VariableEditDistanceAligner(relevantArgs: Set[String]) {
+class VariableEditDistanceAligner(relevantArgs: Set[String] = Set("variable")) {
```
Comment (BeckySharp) [inline text_reading/src/main/scala/org/clulab/aske/automates/alignment/Aligner.scala]: @maxaalexeeva -- seeing this is triggering a memory for me -- there was some function or something where I think at one point in the API refactoring the relevant Args wasn't getting populated right, did I remember to fix it?  Basically, is it working the way we expect for both comment extractions and text extractions?  how can we check?

**Comment by maxaalexeeva** [comment]
Thanks, @BeckySharp and @pauldhein for the quick turnaround!

@pauldhein, it is Levenshtein_dist that I am using. @BeckySharp and @pauldhein, I'll try to plug in the metrics you are proposing. I'll let you know when I am done.

@pauldhein, absolutely agree with having separate endpoints.

**Comment by pauldhein** [comment]
Thanks @maxaalexeeva !

File: text_reading/src/main/scala/org/clulab/aske/automates/alignment/Aligner.scala (hunk 1 above)
Comment (maxaalexeeva) [reply text_reading/src/main/scala/org/clulab/aske/automates/alignment/Aligner.scala]: I'll look into this. Thanks!

**Comment by maxaalexeeva** [comment]
New sample output for the `groundMentionsToSVO` endpoint with the mentions returned for these toy sentences: `... where LAI is the crop canopy. If L falls below that of permanent wilting point (Lpwp)...`:

```{
    "groundings": [
        {
            "variable": "Lpwp",
            "groundings": [
                {
                    "searchTerm": "wilting-point",
                    "osvTerm": "wilting-point",
                    "className": "Attribute",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting",
                    "className": "Process",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "point",
                    "osvTerm": "point",
                    "className": "Mathematical Abstraction",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "point",
                    "osvTerm": "point",
                    "className": "Abstraction",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point",
                    "className": "Attribute",
                    "score": [
                        0.5384615384615384
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point_pressure_head",
                    "className": "Standardized Property",
                    "score": [
                        0.25925925925925924
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point_pressure_head",
                    "className": "Quantitative Property",
                    "score": [
                        0.25925925925925924
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point_pressure_head",
                    "className": "Property",
                    "score": [
                        0.25925925925925924
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point_volume_fraction",
                    "className": "Quantitative Property",
                    "score": [
                        0.2413793103448276
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "wilting",
                    "osvTerm": "wilting-point_volume_fraction",
                    "className": "Property",
                    "score": [
                        0.2413793103448276
                    ],
                    "source": "SVO"
                }
            ]
        },
        {
            "variable": "LAI",
            "groundings": [
                {
                    "searchTerm": "crop_canopy",
                    "osvTerm": "crop_canopy",
                    "className": "Phenomenon",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "crop",
                    "className": "Phenomenon",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Part",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Spatial Phenomenon",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Phenomenon",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "canopy",
                    "osvTerm": "canopy",
                    "className": "Body",
                    "score": [
                        1
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "crop_canopy",
                    "className": "Phenomenon",
                    "score": [
                        0.36363636363636365
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "crop~mature~dry",
                    "className": "Phenomenon",
                    "score": [
                        0.26666666666666666
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "crop~mature~dry_tops",
                    "className": "Phenomenon",
                    "score": [
                        0.2
                    ],
                    "source": "SVO"
                },
                {
                    "searchTerm": "crop",
                    "osvTerm": "(land_allocation)@medium_crop",
                    "className": "Phenomenon",
                    "score": [
                        0.13793103448275862
                    ],
                    "source": "SVO"
                }
            ]
        }
    ]
```
Ended up going with the metric @pauldhein proposed (fewer 0s after the decimal point in the scores). 

@pauldhein, if these outputs look good, would you test out the end point? if everything works, feel free to merge.

**Comment by pauldhein** [comment]
@maxaalexeeva I'm too busy right now to test the integration of the endpoint with a python request, and I am going to have a separate PR that does just that. Given that, and the fact that you already have the updates we requested implemented, I am going to merge this branch.

**Pull request merged**